src/main/java/no/ntnu/mycbr/rest/controller/EphemeralController.java

```diff
@@ -0,0 +1,121 @@
+ * a very short time) case bases.
+ * @author Amar Jaiswal
+ * @since 20 Nov 2019
+ */
+@RestController
+public class EphemeralController {
+
+    private static final String DEFAULT_CASE_ID = "car0";
+    private final Log logger = LogFactory.getLog(getClass());
+
+    //@Autowired
+    //private EphemeralService ephemeralService;
+
+    /**
+     * Perform retrieval on an ephemeral case base with a single query (caseID).
+     * @param conceptID   : Concept name of the myCBR project.
+     * @param casebaseID  : Case base name of the myCBR project.
+     * @param amalgamationFunctionID : Amalgamation function (global similarity function) of the myCBR project.
+     * @param queryCaseID : The caseID that serves as a query
+     * @param k           : Number of desired retrieved cases per query case.
+     * @param ephemeralCaseIDs : The ephemeralCaseIDs is a list of caseIDs used for creating an ephemeral case base.
+     * @return List<Map<String, String>> : List<E> - list of cases (E) from the ephemeral case base. 
+     * The number of cases in the list depends on the value of k. 
+     * Map<K,V> - case representation as a map, where K is the attribute name (including the caseID and similarity) 
+     * as a String and V is the respective value as a String.
+     */
+   @ApiOperation(value = GET_SIMILAR_CASES_FROM_EPHEMERAL_CASE_BASE_WITH_CONTENT, nickname = GET_SIMILAR_CASES_FROM_EPHEMERAL_CASE_BASE_WITH_CONTENT)
+    @RequestMapping(method = RequestMethod.POST, path=PATH_EPHEMERAL_RETRIEVAL_WITH_CONTENT, produces=APPLICATION_JSON)
+    @ApiResponsesDefault
+    public @ResponseBody List<Map<String, String>> retrievalFromEphemeralCaseBaseWithContent(
+	    @PathVariable(value=CONCEPT_ID) String conceptID,
+	    @PathVariable(value=CASEBASE_ID) String casebaseID, 
+	    @PathVariable(value=AMAL_FUNCTION_ID) String amalgamationFunctionID,
+	    @RequestParam(value=CASE_ID, defaultValue=DEFAULT_CASE_ID) String queryCaseID,
+	    @RequestParam(required = false, value=NO_OF_RETURNED_CASES,defaultValue = DEFAULT_NO_OF_CASES) int k,
+	    @RequestBody(required = true)  Set<String> ephemeralCaseIDs) {
+
+	EphemeralService ephemeralService = new EphemeralService(conceptID, casebaseID, amalgamationFunctionID, k);
+	List<Map<String, String>> retrivalResults = ephemeralService
+		.ephemeralRetrivalForSingleQuery(queryCaseID, ephemeralCaseIDs);
+
+	return retrivalResults;
+    }
+    
+    /**
+     * Perform retrieval with multiple queries (caseIDs) on an ephemeral case base.
+     * @param conceptID  : Concept name of the myCBR project.
+     * @param casebaseID : Case base name of the myCBR project.
+     * @param amalgamationFunctionID : Amalgamation function (global similarity function) of the myCBR project.
+     * @param k          : Number of desired retrieved cases per query case.
+     * @param mapOfcaseIDs : Keys are <code>query_case_id_list</code> and <code>casebase_case_id_list</code>. 
```

Comment:
So which of them are keys? Union of these two lists?
What are values?

Reply:
fixed in commit 0607cc6